Files/ViewModels/Bundles/BundleContainerViewModel.cs

```diff
@@ -0,0 +1,286 @@
+﻿using Files.Commands;
+using Files.SettingsInterfaces;
+using Microsoft.Toolkit.Mvvm.ComponentModel;
+using Microsoft.Toolkit.Mvvm.Input;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Windows.Input;
+using Windows.ApplicationModel.DataTransfer;
+using Windows.Storage;
+using Windows.System;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Input;
+
+namespace Files.ViewModels.Bundles
+{
+	/// <summary>
+	/// Bundle's contents view model
+	/// </summary>
+	public class BundleContainerViewModel : ObservableObject, IDisposable
+	{
+		#region Singleton
+
+		private IJsonSettings JsonSettings => associatedInstance?.InstanceViewModel.JsonSettings;
+
+		#endregion
+
+		#region Private Members
+
+		private readonly IShellPage associatedInstance;
+
+		#endregion
+
+		#region Public Properties
+
+		/// <summary>
+		/// A list of Bundle's contents
+		/// </summary>
+		public ObservableCollection<BundleItemViewModel> Contents { get; private set; } = new ObservableCollection<BundleItemViewModel>();
+
+		private string _BundleName = "DefaultBundle";
+		public string BundleName
+		{
+			get => _BundleName;
+			set => SetProperty(ref _BundleName, value);
+		}
+
+		public Visibility NoBundleContentsTextVisibility
+		{
+			get => Contents.Count > 0 ? Visibility.Collapsed : Visibility.Visible;
+		}
+
+		private string _BundleRenameText = string.Empty;
+		public string BundleRenameText
+		{
+			get => _BundleRenameText;
+			set => SetProperty(ref _BundleRenameText, value);
+		}
+
+		private Visibility _BundleRenameVisibility = Visibility.Collapsed;
+		public Visibility BundleRenameVisibility
+		{
+			get => _BundleRenameVisibility;
+			set => SetProperty(ref _BundleRenameVisibility, value);
+		}
+
+		#endregion
+
+		#region Commands
+
+		public ICommand RemoveBundleCommand { get; set; }
+
+		public ICommand RenameBundleCommand { get; set; }
+
+		public ICommand RenameBundleConfirmCommand { get; set; }
+
+		public ICommand RenameTextKeyDownCommand { get; set; }
+
+		public ICommand DragOverCommand { get; set; }
+
+		public ICommand DropCommand { get; set; }
+
+		#endregion
+
+		#region Constructor
+
+		public BundleContainerViewModel(IShellPage associatedInstance)
+		{
+			this.associatedInstance = associatedInstance;
+			this.BundleRenameText = BundleName;
+
+			// Create commands
+			RemoveBundleCommand = new RelayCommand(RemoveBundle);
+			RenameBundleCommand = new RelayCommand(RenameBundle);
+			RenameBundleConfirmCommand = new RelayCommand(RenameBundleConfirm);
+			RenameTextKeyDownCommand = new RelayParameterizedCommand((e) => RenameTextKeyDown(e as KeyRoutedEventArgs));
```

Comment:
`RelayCommand` already supports parameters. So no need to create separate type for this.
```csharp
RenameTextKeyDownCommand = new RelayCommand<KeyRoutedEventArgs>(RenameTextKeyDown);
```

Reply:
:eyes: